lib/decimal.ex

```diff
@@ -731,6 +731,33 @@ defmodule Decimal do
   end
 
   @doc """
+  Parses a string into a decimal.
+
+  If successful, returns a tuple in the form of `{:ok, decimal}.
+  Otherwise `:error`.
+
+  ## Examples
+
+      iex> Decimal.parse("3.14")
+      {:ok, Decimal.new(3.14)}
+
+      iex> Decimal.parse("-1.1e3")
+      {:ok, Decimal.new(-1.1e3)}
+
+      iex> Decimal.parse("bad")
+      :error
```

Comment:
we could have a richer errors with `{:error, ...}`, however I kept it as just atom to be consistent with `{Integer,Float,Version}.parse/1`